### ISSUE TITLE
Leveraging the Apache configencoder to generate Fluentd configuration files

### ISSUE THREAD

**Issue by pedrojflores** [jtyr/ansible-config_encoder_filters#6]
I've been trying to rely on the apache configencoder to generate Fluentd configuration files and I'm running into an issue with how sections are rendered on the resulting configuration file.  

The following block:
```
td_agent_config__custom:
  content:
    - sections:
      - name: source
        content:
          - options:
            - source_hostname_key: source_host
            - '@type': forward
            - port: 24224
      - name: filter
        content:
          - options:
            - '@type': record_transformer
          - sections:
            - name: record
              content:
                - options:
                  - source: "${hostname}"
```

Outputs the following 

```
<source >
  source_hostname_key source_host
  @type forward
  port 24224
</source>

<filter >
  @type record_transformer

  <record >
    source ${hostname}
  </record>
</filter>
```

Notice the spaces at the end of each opening section.  Fluentd doesn't like those spaces.  This is most likely due to the fact that Fluentd, as opposed to Apache, does not always require a 'param' to be present in each section.  

What I need is the following output
```
<source>
  source_hostname_key source_host
  @type forward
  port 24224
</source>

<filter>
  @type record_transformer

  <record>
    source ${hostname}
  </record>
</filter>
```

Is there a way to make the apache config encoder support this use case?  In the interim, on my end, I've made a copy of the apache config encoder, renamed it to encode_fluentd with the changes I need for it to work they way I need it to.  Any suggestions on how to better deal with this use case will be greatly appreciated.

**Comment by jtyr** [comment]
We would have modify the encoder to check if there is any parameter and if not, don't print the space. If you have time, please create a PR for that.

**Comment by jtyr** [comment]
Fixed via PR #9.
